### PR TITLE
 Remove CLI usage in Vert.x Core (examples and code)

### PR DESCRIPTION
**Pull request by tsegismont**
Commits can be reviewed individually

### CODE REVIEW

**Comment by vietj** [comment]
it seems this PR removes more than just examples, so I'm guessing we need to wait until this is resurrected in vertx-launcher to merge it to avoid vertx-shell breaking

File: src/main/java/io/vertx/core/AbstractVerticle.java [hunk 1]
```diff
@@ -87,10 +88,13 @@ public JsonObject config() {
 
   /**
    * Get the arguments used when deploying the Vert.x process.
+   *
```
Comment (vietj) [inline src/main/java/io/vertx/core/AbstractVerticle.java:91]: the Java doc should be changed to : "returns an empty list"

File: src/main/java/io/vertx/core/Context.java [hunk 2]
```diff
@@ -155,10 +155,11 @@ static boolean isOnVertxThread() {
   @Nullable JsonObject config();
 
   /**
-   * The process args
+   * @deprecated As of version 5, Vert.x is no longer tightly coupled to the CLI
```
Comment (vietj) [inline src/main/java/io/vertx/core/Context.java:158]: should also say that now it returns an empty list

**Review by vietj — CHANGES_REQUESTED** [review]
Couple of minor changes, but LGTM

**Comment by tsegismont** [comment]
@vietj done

**Pull request merged**